src/views/catalog/customize/components/CustomizeForms/useCustomizeFormSubmit.ts

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { getAnnotation } from '@kubevirt-utils/resources/shared';
+import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template/utils/selectors';
+
+import { useWizardVMContext } from '../../../utils/WizardVMContext';
+import { processTemplate } from '../../utils';
+
+type useCustomizeFormSubmitType = [
+  onSubmit: (event: any) => Promise<void>,
+  loaded: boolean,
+  error: any,
+];
+
+export const useCustomizeFormSubmit = (template: V1Template): useCustomizeFormSubmitType => {
+  const { ns } = useParams<{ ns: string }>();
+  const history = useHistory();
+  const [templateLoaded, setTemplateLoaded] = React.useState(true);
+  const [templateError, setTemplateError] = React.useState<any>();
+
+  const { updateVM, loaded: vmLoaded, error: vmError } = useWizardVMContext();
+
+  const onSubmit = async (event) => {
+    event.preventDefault();
+    setTemplateLoaded(false);
+    try {
+      const formData = new FormData(event.currentTarget as HTMLFormElement);
+
+      const processedTemplate = await processTemplate(template, ns, formData);
+
+      const vm = getTemplateVirtualMachineObject(processedTemplate);
+
+      vm.metadata.annotations[ANNOTATIONS.displayName] = getAnnotation(
```

Comment:
that looks like a bug, the vm display name is not the template display name
a - the requirement of this Task dose not talk abou it
b - did you check with design about that ?

Reply:
@glekner

Reply:
IMHO @glekner just copied this bug from our old code ...
it does not make sense to me that all VMs created from the same template will have the same display name, am I missing something ?

Reply:
btw/ the reason we need a new code instead of just copy the old one is all this little bugs that are hiding in helpers ans setters in the old code